### PR TITLE
New user administration page

### PR DESCRIPTION
**Pull request by Cheshiriks**
- added new user administration page
- added new user status

Снимок

### CODE REVIEW

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/userprofile/UserProfileView.kt [hunk 1]
```diff
@@ -86,15 +87,26 @@ val userProfileView: FC<UserProfileViewProps> = FC { props ->
         // ===================== RIGHT COLUMN =======================================================================
         div {
             className = ClassName("col-6 mb-4 mt-2")
-            tab(selectedMenu.name, UserProfileTab.values().map { it.name }, "nav nav-tabs mt-3") { value ->
-                setSelectedMenu { UserProfileTab.valueOf(value) }
+            props.currentUserInfo?.globalRole?.let { role ->
+                val tabList = if (role.isHigherOrEqualThan(Role.SUPER_ADMIN)) {
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/userprofile/UserProfileView.kt:91]:
```suggestion
                val tabList = if (role.isSuperAdmin()) {
```

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/userprofile/UserProfileView.kt [hunk 2]
```diff
@@ -86,15 +87,26 @@ val userProfileView: FC<UserProfileViewProps> = FC { props ->
...
             }
 
             when (selectedMenu) {
                 UserProfileTab.VULNERABILITIES -> renderVulnerabilityTableForProfileView {
-                    this.userName = userName
                     this.vulnerabilities = vulnerabilities
                 }
+                UserProfileTab.USERS -> renderNewUsersTableForProfileView {
+                    this.userName = userName
+                    // this.vulnerabilities = vulnerabilities
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/userprofile/UserProfileView.kt:108]:
```suggestion
```

**Review by sanyavertolet — APPROVED** [review]

**Review by nulls — APPROVED** [review]

**Review by kgevorkyan — APPROVED** [review]

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/userprofile/UserProfileView.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/userprofile/UserProfileView.kt]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/userprofile/UserProfileView.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/userprofile/UserProfileView.kt]: Fixed

**Comment by akuleshov7** [comment]
Approvals for new users and moderation

We also need to support following: 
- [ ] Source should be a link (not a string, where we can construct it) ?
- [ ] On registration page ask user to set some public link to them (linkedin, GitHub, gitee, etc.). 
- [ ] On registration page mention, that now we have limited resources, so we are selecting people and not approving the access to write functionality for everyone.
- [ ] On registration page add some small description field "Please add some information about you, so it will be easier to approve for us"

**Review by nulls — APPROVED** [review]

**Pull request merged**